node_settled_handler.go

```diff
@@ -115,25 +110,60 @@ func (p *NodeSettledHandler) subscribeEvents(ctx context.Context) error {
 		if !ok {
 			continue
 		}
-		if !finalEvent.FinalHtlcEvent.Settled {
-			// TODO: Handle final fail?
-
-			continue
-		}
 
 		key := types.CircuitKey{
 			ChanID: event.IncomingChannelId,
 			HtlcID: event.IncomingHtlcId,
 		}
 
-		if err := p.handleFinalHtlc(ctx, key); err != nil {
+		failedResolution := !finalEvent.FinalHtlcEvent.Settled
+
+		if err := p.handleFinalHtlc(ctx, key, failedResolution); err != nil {
 			return err
 		}
 	}
 }
 
 func (p *NodeSettledHandler) handleFinalHtlc(ctx context.Context,
-	key types.CircuitKey) error {
+	key types.CircuitKey, failedResolution bool) error {
```

Comment:
Mostly `settled bool` is used, but this is `failed bool`. Maybe better to keep it consistent and always use `settled`.